### PR TITLE
ci: migrating to GitHub actions

### PR DESCRIPTION
**Pull request by JBAhire**
## Description
Migrating to GitHub actions from CircleCI as per https://github.com/hypertrace/hypertrace/issues/144

### Checklist:
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Any dependent changes have been merged and published in downstream modules

### Documentation
- https://docs.github.com/en/free-pro-team@latest/actions

### CODE REVIEW

**Review by mohit-a21 — APPROVED** [review]

File: .github/workflows/pr-test.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+          make deps
+          make deps-ci
+      
+      - name: golangci-lint
+        uses: golangci/golangci-lint-action@v2
+        with:
+          # Required: the version of golangci-lint is required and must be specified without patch version: we always use the latest patch version.
+          version: v1.29
+
+      - name: run tests
```
Comment (jcchavezs) [inline .github/workflows/pr-test.yml:37]:
```suggestion
      - name: Run tests
```

File: .github/workflows/pr-test.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+      - name: Setup go
+        uses: actions/setup-go@v1
+        with:
+          go-version: ${{ matrix.go }}
+      
+      - name: download dependencies
+        run: |
+          make deps
+          make deps-ci
```
Comment (jcchavezs) [inline .github/workflows/pr-test.yml:29]:
```suggestion

```

File: .github/workflows/pr-test.yml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+      
+      - name: golangci-lint
```
Comment (jcchavezs) [inline .github/workflows/pr-test.yml:31]:
```suggestion
      - name: Lint files
```

File: .github/workflows/pr-test.yml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+      - name: Upload coverage to Codecov
+        uses: codecov/codecov-action@v1
+        with:
+          name: go-agent test reports
+          fail_ci_if_error: true
+          files: ./coverage.txt
+
+      - name: check examples
+        run: make check-examples
```
Comment (jcchavezs) [inline .github/workflows/pr-test.yml:48]:
```suggestion
        run: make check-examples
        
```

File: .github/workflows/pr-test.yml [hunk 5]
```diff
@@ -0,0 +1,48 @@
+name: test
+on:
+  push:
+    branches:
+      - main
```
Comment (jcchavezs) [inline .github/workflows/pr-test.yml:5]:
```suggestion
      - main
    paths-ignore:
      - "**/*.md"
      - "LICENSE"
```

**Review by jcchavezs — APPROVED** [review]
Great work!

**Pull request merged**